util/fibers/detail/fiber_interface.cc

```diff
@@ -267,23 +268,27 @@ ctx::fiber_context FiberInterface::SwitchTo() {
 
   auto& fb_initializer = FbInitializer();
   std::swap(fb_initializer.active, prev);
-  ++fb_initializer.epoch;
-  uint64_t tsc = CycleClock::Now();
-
-  DCHECK_GE(tsc, cpu_tsc_);
-  DCHECK_GE(tsc, prev->cpu_tsc_);
-
-  fb_initializer.switch_delay_cycles += (tsc - cpu_tsc_);
-  cpu_tsc_ = tsc;
 
-  // prev tsc points to the fiber that was active before this call.
-  uint64_t delta_cycles = tsc - prev->cpu_tsc_;
-  if (delta_cycles > g_tsc_cycles_per_ms) {
-    fb_initializer.long_runtime_cnt++;
+  uint64_t tsc = CycleClock::Now();
 
-    // improve precision, instead of "delta_cycles / (g_tsc_cycles_per_ms / 1000)"
-    fb_initializer.long_runtime_usec += (delta_cycles * 1000) / g_tsc_cycles_per_ms;
+  // When a kernel suspends we may get a negative delta because TSC is reset.
+  // We ignore such cases (and they are very rare).
+  if (tsc > cpu_tsc_) {
```

Comment:
I don't know how the reset is done, so it might be a stupid question, but shouldn't we do `cpu_tsc_ = tsc` regardless of the `if` result (i.e. even if it's `<=`)?

Reply:
not stupid at all 🤦